packages/plugin-ext/src/main/browser/terminal-main.ts

```diff
@@ -44,6 +44,9 @@ export class TerminalServiceMainImpl implements TerminalServiceMain, Disposable
         }
         this.toDispose.push(this.terminals.onDidChangeCurrentTerminal(() => this.updateCurrentTerminal()));
         this.updateCurrentTerminal();
+        this.terminals.getDefaultShell().then(shell => {
```

Comment:
Could not it be a race? If an extension reads `shell` before this method resolves?

Probably the better place to pass it here during initialization: https://github.com/svor/theia/blob/0e002ce2d8baa8fe9a90245912fa20e412d48e2c/packages/plugin-ext/src/hosted/browser/hosted-plugin.ts#L403

Reply:
makes sense, moved the initialization into hosted-plugin.